### ISSUE TITLE
[Github Action] upload-sarif flag usage

### ISSUE THREAD

**Issue by alarkirikal** [google/osv-scanner#1454]
Does the `upload-sarif` flag work as expected?

I am using a very simple github action .yml mostly taken off the documentation website (https://google.github.io/osv-scanner/github-action/#scan-on-pull-request)

```
name: OSV-Scanner PR Scan
on: [ pull_request ]

concurrency:
  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
  cancel-in-progress: true

permissions:
  actions: read
  security-events: write
  contents: read

jobs:
  scan-pr:
    uses: "google/osv-scanner-action/.github/workflows/osv-scanner-reusable.yml@v1.9.1"
    with:
      scan-args: |-
        --upload-sarif=false
```

Note that I do not want to upload sarif for the code-scanning feature on Github. I just want to run the action, get the results and not have it mapped to Advanced Security features of Github.

This still tries to run "Upload to code-scanning" and throws an error:
```
Run github/codeql-action/upload-sarif@...
Warning: Advanced Security must be enabled for this repository to use code scanning.
Uploading results
  Processing sarif files: ["results.sarif"]
  Validating results.sarif
  Combining SARIF files using the CodeQL CLI
  Adding fingerprints to SARIF file. See https://docs.github.com/en/enterprise-cloud@latest/code-security/code-scanning/integrating-with-code-scanning/sarif-support-for-code-scanning#providing-data-to-track-code-scanning-alerts-across-runs for more information.
  Uploading results
  Warning: Advanced Security must be enabled for this repository to use code scanning.
  Error: Advanced Security must be enabled for this repository to use code scanning.
  Warning: Advanced Security must be enabled for this repository to use code scanning.
```

How do I properly disable uploading to code-scanning? Seems like it's ignoring the scan-args that I'm setting.

**Comment by another-rex** [comment]
**tldr:**

Replace: 
```
...
    with:
      scan-args: |-
        --upload-sarif=false
...
```

with: 

```
...
    with:
      upload-sarif: false
...
```
---

**Explaination:**

You are passing `upload-sarif` wrong here, 

> scan-args: This value is passed to osv-scanner CLI 

But upload-sarif is an input to the github action, so you'll want to pass it directly under `with:`, i.e. at the same level as `scan-args`.

Also, if you want to use scan args, you need to be careful to also pass in the default value as well. The version you have above does not specify anything to actually scan. E.g. Something like the following:

```
scan-args: |-
  # Your flags here
  --recursive # Recursively scan subdirectories
  --skip-git=true # Skip commit scanning to focus on dependencies
  ./ # Start the scan from the root of the repository
```

---

Something actionable here for us is probably to update the documentation with an example of passing in `upload-sarif: false`. Let me know if there's any other improvements to the documentation you can think of to better clarify this.

**Comment by alarkirikal** [comment]
Ah how silly of me, I'm sorry! You're right and works nicely as suggested. Thank you.
